src/sentry/api/endpoints/rpc.py

```diff
@@ -28,14 +33,37 @@ def post(self, request: Request, service_name: str, method_name: str) -> Respons
             raise PermissionDenied
 
         metadata = request.data.get("meta")  # noqa
-        arguments = request.data.get("args")
+        try:
+            arguments: Dict[str, Any] = request.data["args"]
+        except KeyError as e:
+            raise ParseError from e
+        if not isinstance(arguments, dict):
+            raise ParseError
+
+        auth_context: AuthenticationContext = AuthenticationContext()
+        if auth_context_json := arguments.get("auth_context"):
+            try:
+                # Note -- generally, this is NOT set, but only in cases where an RPC needs to invoke code
+                # that depends on the `env.request.user` object.  In that case, the authentication context
+                # includes an authenticated user that will be injected into the global request context
+                # for compatibility.  Notably, this authentication context is *trusted* as the request comes
+                # from within the privileged RPC channel.
```

Comment:
It's not possible generally though.  In a lot of places we are accessing this value inside of an ORM helper method that is easily 3-4 levels beyond the controller.  I think it will get messy fast.